src/Moryx.Runtime.Endpoints/Modules/Endpoint/ModulesController.cs

```diff
@@ -0,0 +1,313 @@
+                    StartBehaviour = _moduleManager.BehaviourAccess<ModuleStartBehaviour>(module).Behaviour,
+                    FailureBehaviour = _moduleManager.BehaviourAccess<FailureBehaviour>(module).Behaviour,
+                    Notifications = notifications.Select(n => new NotificationModel(n)).ToArray()
+                };
+
+                var dependencies = _moduleManager.StartDependencies(module);
+                foreach (var dependency in dependencies)
+                {
+                    model.Dependencies.Add(new ServerModuleModel
+                    {
+                        Name = dependency.Name,
+                        HealthState = dependency.State
+                    });
+                }
+                models.Add(model);
+            }
+            return models;
+        }
+
+        [HttpGet("module/{moduleName}/healthstate")]
+        public ActionResult<ServerModuleState> HealthState([FromRoute] string moduleName)
+        {
+            var module = _moduleManager.AllModules.FirstOrDefault(m => m.Name == moduleName);
+            if (module != null)
+                return module.State;
+
+            Response.StatusCode = (int)HttpStatusCode.NotFound;
+            return ServerModuleState.Failure;
+        }
+
+        [HttpGet("module/{moduleName}/notifications")]
+        public ActionResult<IEnumerable<NotificationModel>> Notifications([FromRoute] string moduleName)
+        {
+            var module = _moduleManager.AllModules.FirstOrDefault(m => m.Name == moduleName);
+            if (module != null)
+                return Ok(module.Notifications.Select(n => new NotificationModel(n)));
+
+            Response.StatusCode = (int)HttpStatusCode.NotFound;
+            return Ok(Enumerable.Empty<NotificationModel>());
+        }
+
+        [HttpPost("module/{moduleName}/start")]
+        public ActionResult Start([FromRoute] string moduleName)
+        {
+            var module = GetModuleFromManager(moduleName);
+            if (module == null)
+                throw new ArgumentException(nameof(moduleName));
+
+            _moduleManager.StartModule(module);
+            return Ok();
+        }
+
+        [HttpPost("module/{moduleName}/stop")]
+        public ActionResult Stop([FromRoute] string moduleName)
+        {
+            var module = GetModuleFromManager(moduleName);
+            if (module == null)
+                throw new ArgumentException(nameof(moduleName));
+
+            _moduleManager.StopModule(module);
+            return Ok();
+        }
+
+        [HttpPost("module/{moduleName}/reincarnate")]
+        public ActionResult Reincarnate([FromRoute] string moduleName)
+        {
+            var module = GetModuleFromManager(moduleName);
+            if (module == null)
+                throw new ArgumentException(nameof(moduleName));
+
+            new Thread(() =>
+            {
+                _moduleManager.ReincarnateModule(module);
+            }).Start();
```

Comment:
This needs to be a background thread and must be disposed after completion. Preferably use ThreadPool or TaskFactory that delegates to the ThreadPool.